File: src/main/java/BoxTester/BoxTester.java
package BoxTester;

import java.util.Scanner;

public class BoxTester {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        float length, width;
        int height;
//        Box3 box3 = new Box3();
/*        Box5 box5 = new Box5();*/
        while(true){
            System.out.print("Please enter object's length: ");
            length = scanner.nextFloat();
            System.out.print("Please enter object's width: ");
            width = scanner.nextFloat();
            System.out.print("Please enter object's height: ");
            height = scanner.nextInt();
            Box box5 = new Box5(length, width, height); //Box 使用 Box5的長寬高
            Box box3 = new Box3(length, width, height); //Box 使用 Box3的長寬高
            if(length == -1||width == -1|| height == -1)
                break;
            if(length > box5.length)
                System.out.println("Please enter again.");
            else if(box3.validate(length, width, height))
                System.out.println("Use Box3");
            else if(box5.validate(length, width, height))
                System.out.println("Use Box5");
            else System.out.println("Please enter again.");

           /* if(length > box5.length)
                System.out.println("Please enter again.");
            else if(box3.validate(length, width, height)) {
                System.out.println("Box3");
            }else if (box5.validate(length, width, height)){
                System.out.println("Box5");
            }else System.out.println("Please enter again.");
*/
        }
    }
}

File: src/main/kotlin/box/BoxTester.kt
package box

import java.util.*

fun main() {
    val scanner = Scanner(System.`in`)
    var length:Float
    var width:Float
    var height:Int
    val box3 = Box3()
    val box5 = Box5()
    while(true){
        print("Please enter object's length: ")
        length = scanner.nextFloat()
        print("Please enter object's width: ")
        width = scanner.nextFloat()
        print("Please enter object's height: ")
        height = scanner.nextInt()
        if(length == -1f || width == -1f || height == -1)
            break
        if(length > box5.length)
            println("Please enter again.")
        else if(box3.validate(length, width, height))
            println("Box3")
        else if(box5.validate(length, width, height))
            println("Box5")
        else println("Please enter again.")


    }

}

class Box3(){
    val length = 23f
    val width = 14f
    val heigth = 13
    fun validate(length:Float, width:Float, heigth:Int) :Boolean{
        if(length < this.length && width < this.width && heigth < this.heigth)
            return true
        else return false
    }


}

class Box5(){
    val length = 39.5f
    val width = 27.5f
    val heigth = 23
    fun validate(length:Float, width:Float, heigth:Int) :Boolean{
        if( length< this.length && width< this.width && heigth< this.heigth)
            return true
        else return false
    }

}